src/code/z_player_lib.c

```diff
@@ -810,20 +878,21 @@ void func_8008F87C(GlobalContext* globalCtx, Player* this, SkelAnime* skelAnime,
             temp3 = func_80041D4C(&globalCtx->colCtx, sp88, sp84);
 
             if ((temp3 >= 2) && (temp3 < 4) && !SurfaceType_IsWallDamage(&globalCtx->colCtx, sp88, sp84)) {
-                footprintPos.y = sp80;
+                footprintPos.y = posB;
                 EffectSsGFire_Spawn(globalCtx, &footprintPos);
             }
         }
     }
 }
 
-s32 func_8008FCC8(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot, void* thisx) {
+s32 Player_OverrideLimbDrawImpl(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot,
+                                void* thisx) {
     Player* this = (Player*)thisx;
 
     if (limbIndex == PLAYER_LIMB_ROOT) {
-        D_80160014 = this->leftHandType;
-        D_80160018 = this->rightHandType;
-        D_80160000 = &this->swordInfo[2].base;
+        sLeftHandType = this->leftHandType;
+        sRightHandType = this->rightHandType;
+        sPartsPosPtr = &this->swordInfo[2].base;
```

Comment:
```suggestion
        sCurLimbPartPos = this->bodyPartsPos - 1;
```
with that name, I'm trying to reflect that `sCurLimbPartPos` is set in `OverrideLimbDraw` and used in `PostLimbDraw` when drawing each limb